### PR TITLE
Make NPM ready

### PR DESCRIPTION
**Pull request by pr1ntr**
Added a package.json as well as browserify bundling and uglify compression.

### CODE REVIEW

**Comment by pr1ntr** [comment]
i made it so that dist is where the useable file is not src. define gets defined by the browserify build. You can submit this to npm and then others can npm install js-handtracking and it will work with their browserify builds. Or they can just clone the repo and use the dist folder.

**Comment by pr1ntr** [comment]
wait on this actually. I need to change the global name

**Comment by pr1ntr** [comment]
OK updated global name to HT. if you just include the script as is HT is available globally. or if you use browserify its var HT = require("HT").

**Comment by pr1ntr** [comment]
err actually require('js-handtracking') not HT.. HT is global name

File: src/handtracking.js [hunk 1]
```diff
@@ -1,154 +1,160 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+var CV = require('./cv.js');
```
Comment (jcmellado) [inline src/handtracking.js]: So, the users of the library need to include "requireJS" in the html file... hmmm, not really what I wanted

**Comment by pr1ntr** [comment]
user doesn't need anything to use what is in the dist folder. Try it yourself. put it in your examples

**Comment by pr1ntr** [comment]
Not sure if I explained this well. What I did was make it so what is in src/ should not be used without some kind of UMD wrapper (ie, require, amd, common, browserify) This is why I set up the npm run build which will build those files in /src into the /dist folder to create something universally useable. If a user just wants to download the script to use in their project without any kind of dependency/package manager (ie, bower or npm) they use those files in /dist and they will work inside a script tag no problem.

**Comment by jcmellado** [comment]
Yes, but I really want to avoid external dependencies. Maybe (not tested, just borrowed):

```
if (typeof define === 'function' && define.amd) { // Not really sure about this first "if"
    define( 'js-handtracking', HT ); // Is "js-handtracking" the right name?
} else if ('undefined' !== typeof exports && 'undefined' !== typeof module) {
    module.exports = HT;
}
```

The examples must work just copying the src files to the samples folder, without modifying the existing html files. This must sound a bit silly, I know, but I have a lot of problems in the past with people spending more time trying to solve external dependencies that having fun with the code.

Thanks for your effort.

**Comment by pr1ntr** [comment]
Ok well this is certainly your prerogative. However if you choose not make your library accessible to people who use more modern web development techniques (i.e., browserify). I will just host my own version of this on github and npm. All credit will be given where credit is due of course.I hope you have no issue with this.

That said, the files that are built in the dist folder have no dependencies. You don't want people to use the src files. The whole point is to generate /dist from /src.

**Comment by jcmellado** [comment]
I understand the benefits, don't worry about that.

What I trying to explain (my English is bad, sorry) is that I don't want to add dependencies to the /src folder. To use "require/modules/..." should be an option, not a requirement.

Take a look to a well-known library like three.js and the way it uses "define" and "modules.exports" only if they are defined to avoid dependencies:
https://github.com/mrdoob/three.js/blob/master/src/Three.js#L9

And take a look too to the way that the three.js team modify a library that I wrote to "browserify" it:
https://github.com/mrdoob/three.js/blob/master/examples/js/loaders/ctm/ctm.js

I'm missing that kind of "if" in the PR, ok?

Anyway, the library is under the MIT license, so feel free to do what you want with it, not problem. :)

**Pull request closed without merging**